### ISSUE TITLE
Dotnetcore local invoke support

### ISSUE THREAD

**Issue by napalm684** [serverless/serverless#4815]
# This is a Feature Proposal

## Local invoke not supported for dotnetcore

For bug reports:
* What went wrong?
* What did you expect should have happened?
* What was the config you used?
* What stacktrace or error message from your provider did you see?

For feature proposals:
* What is the use case that should be solved. The more detail you describe this in the easier it is to understand for us. 
_sls invoke local -f <function _name>_ should work as it does with other runtimes.
* If there is additional config how would it look

Similar or dependent issues:
* #12345

## Additional Data

* ***Serverless Framework Version you're using***:
1.26.1
* ***Operating System***:
Ubuntu 16.04
* ***Stack Trace***:
* ***Provider Error messages***:
You can only invoke Node.js, Python & Java functions locally

**Comment by horike37** [comment]
Thank you for the proposal @napalm684 :+1: 
yes! the functionality would be needed, so I just added `help wanted` tag so that someone can pick it up.

**Comment by mikeweltejr** [comment]
If this is still an issue I'd like to pick it up and start working on it? Let me know.

**Comment by horike37** [comment]
@mikeweltejr 
Awesome! This is still an issue.
Let us know if you have any question or help :+1:

**Comment by mikeweltejr** [comment]
Sure! So looking at this it seems when you create a new aws-csharp project it comes with sdk 1.0.4, running this locally would need to assume the user would have this version of the sdk on their local machine in order to run it, which might be a bold assumption (for instance I have v2.1.104 sdk). From my understanding that is a bit different from nodejs since with node it wouldn't matter which version you used to create you can still simply run node fileName.js and it will run it. Let me know if you have an acceptable path forward for this, in the meantime I'm going to look to see if there are other issues I could pick up.

**Comment by horike37** [comment]
@mikeweltejr 
Thank you for in-depth consideration :+1: 

Imo, you wouldn't need the assumption since it would be hard to provide the same version corresponding to users environment in the new aws-csharp template.
Users should change that as needed. What do you think?

**Comment by mikeweltejr** [comment]
@horike37 yeah that would work, I wanted to ensure that was an acceptable resolution. So I will first ensure this works with the version I have, then test out other versions as well. I will add to the documentation when working to note that the sdk version in globals.json of their project needs to match their local environment sdk in order to invoke locally. Are there any sdk versions that serverless does NOT currently support for c#? Mainly the dotnetcore 2.x stuff?

**Comment by horike37** [comment]
@mikeweltejr 
>I will add to the documentation when working to note that the sdk version in globals.json of their >project needs to match their local environment sdk in order to invoke locally.

That's a good idea :+1: 

>Are there any sdk versions that serverless does NOT currently support for c#? Mainly the dotnetcore >2.x stuff?

 Ah I remember now, aws-csharp template has already updated for 2.x stuff on #4656, so it will be mainly in next Serverless version.

**Comment by fbeauchemin** [comment]
@mikeweltejr do you have more details on the current implementation / status on this feature?

**Comment by safv12** [comment]
Is there any update about this feature?
